### PR TITLE
Refactor houses controller and helper

### LINKED ISSUE #49: Fixing coding style for Houses
Hey @hoangvq288,

Your work on PR #33 was awesome. I just wanted you to remember it. The PR was about fixing coding style for `HousesHelper.rb` and `HousesController.rb`.

> Also, I'm looking a contributor who can get rid off linter warnings.

### PR DESCRIPTION
**Pull request by florinionce**
This PR resolves #49.
* improve codestyle
* keep it dry
* remove unnecessary callback

### CODE REVIEW

**Comment by BerkhanBerkdemir** [comment]
Thank you @florinionce,

This is awesome. Today I learned a lot with this PR. I will review it for today.

**Comment by BerkhanBerkdemir** [comment]
@NARKOZ, you all did almost same thing, and I stated that because the code seems more readable. On the other hand, I didn't say anything bad about your code. Both are same, and I need to decide it. At the same time, I said on https://github.com/magara/magara/pull/52#issuecomment-431085365

> However, I love [that line](https://github.com/magara/magara/pull/52/files#diff-fb0825b6ddd7cbef5bbe25d51689ba2aR19), so can we use that line for PR #53?

**Comment by florinionce** [comment]
Oh, sorry @NARKOZ - I didn't see that there was already a PR for this issue.

**Comment by BerkhanBerkdemir** [comment]
Do you know to add an author name/email, @florinionce? If you don't, here is [a guide](https://help.github.com/articles/creating-a-commit-with-multiple-authors/). Also, I don't know why, but `[Skip CI]` doesn't work with Semaphore CI.

File: app/helpers/houses_helper.rb [hunk 1]
```diff
@@ -22,18 +19,6 @@ def int_to_gender(int)
...
-              '&cross;'
-    end
-
-    value.html_safe
-  end
-
-  # TODO: DRY it. HousesController has same method.
-  def owner?
-    true if @house.user == current_user
+    value == true ? '&checkmark;'.html_safe : '&cross;'.html_safe
```
Comment (BerkhanBerkdemir) [inline app/helpers/houses_helper.rb]: Value will be boolean (it comes from Active Record), so we don't need to use `== true`.

```suggestion
    value ? '&checkmark;'.html_safe : '&cross;'.html_safe
```

File: app/controllers/houses_controller.rb [hunk 2]
```diff
@@ -1,7 +1,8 @@
 class HousesController < ApplicationController
   before_action :authenticate_user!, except: %i[index show]
   before_action :set_house, only: %i[show edit update destroy]
-  before_action :owner?, only: %i[edit update destroy]
```
Comment (BerkhanBerkdemir) [inline app/controllers/houses_controller.rb]: Why did you delete that line? Only owner can edit or delete a property.

**Comment by BerkhanBerkdemir** [comment]
Is there any update, @florinionce? Let me know if something goes wrong.

**Comment by florinionce** [comment]
@BerkhanBerkdemir Sorry for the delay, I've been on holiday and I didn't check my email 👍

**Comment by BerkhanBerkdemir** [comment]
No worry, and thanks, @florinionce! However, could you show us the reason for deleting [ app/controllers/houses_controller.rb#4](https://github.com/magara/magara/pull/53#discussion-diff-226760738L1) `before_action` because I didn't get it -didn't see the full picture-.

File: app/controllers/houses_controller.rb (hunk 2 above)
Comment (florinionce) [reply app/controllers/houses_controller.rb]: Well, it seemed that it's not doing anything, since the `owner?` method is only returning a boolean. If it returns false (the user is not an owner), it's not redirecting the user, or raise an unauthorized exception. Or am I missing something?

Comment (BerkhanBerkdemir) [reply app/controllers/houses_controller.rb]: What if a normal user deletes or accesses for details of a property? How will we control it? I was trying to say "only owner can access edit and destroy pages". Does it make sense? You're right about guest users; however, we need to handle normal users.

Comment (florinionce) [reply app/controllers/houses_controller.rb]: Well, I agree. But I didn't delete the `owner?` method - I just deleted a callback which returned a boolean. It wasn't restricting the guest users from accessing anything.

Comment (BerkhanBerkdemir) [reply app/controllers/houses_controller.rb]: I'm aware of signed users who are not the owner of property. Hmm, I will look at afternoon. But thanks for discussing that.

Comment (BerkhanBerkdemir) [reply app/controllers/houses_controller.rb]: I downloaded your branch and test it. Please try on your environment these steps

1. Sign up with a user (lorem@example.com)
2. Submit a house with details
3. Log out from lorem@example.com
4. Sign up with other user (ipsum@example.com)
5. Find the house which we submitted with lorem@example.com
6. You won't see any button about edit or delete, but you can go to the edit page. Enter an URL. For example http://localhost:3000/houses/1 to http://localhost:3000/houses/1/edit
7. You will see the edit page, and also you can delete it.

So, we can not delete that callback. Does that make sense?
